### PR TITLE
Exorcise the BMC and access GPIOs through sysfs

### PR DESCRIPTION
**Pull request by jonasmalacofilho**
Hi Joshua!

In the last couple of days I assembled and started testing a RPiBMC prototype to handle LOM for a machine that I now need to remotely (and reliably) access.

I closely followed your 2017 post (most notable change was adding resistors to limit the current at the bases of the transistors), and everything went very smoothly.  Thanks!

I did find a few issues on the software side though.  Most notably the daemon kept the CPU at 100%, which caused severe contention on the old original Model B RPi I'm using for this.  

So I ported the daemon to Python 3, as well as replaced the obvious `while True: pass` hot loop.  Still, CPU utilization remained high when using `daemon_runner.do_action()` (but not when directly calling `app.run()`).  

In addition to this, I had the daemon cause an unexpected shutdown of the host system, and another time the daemon became totally unresponsive.

No useful information about those failures was logged, but I started to poke around the sysfs interface for the GPIOs and I was able to confirm that the hardware and kernel were still working fine.

This gave me the idea to make the shell directly access the relevant GPIOs through that sysfs interface, eliminating the need for a daemon.

This obviously doesn't work for the locate function, but I have admit I don't see the point of it (and I haven't built that part of the circuit).  And even if it were to be implemented, the auxiliary process (probably) required for it could (and IMO should) be kept separate from the more critical features of host state and power/reset switches.

Long story short, this is a (draft) pull-request with all changes I've made so far, in case you find any of them interesting and/or useful.

### CODE REVIEW

**Comment by joshuaboniface** [comment]
Thanks for the effort! Unfortunately I have effectively abandoned this one, since I never got the serial console aspect working right and ended up upgrading the servers that needed this back in 2019.
